apps/api/src/services/objectIndex.service.ts

```diff
@@ -0,0 +1,301 @@
+import { eq } from "drizzle-orm";
+import { db } from "@/database/db.js";
+import { objectIndex } from "@/database/schema/syncing/defs.js";
+import { trades } from "@/database/schema/trading/defs.js";
+import { agentRankHistory } from "@/database/schema/ranking/defs.js";
+import { competitionsLeaderboard } from "@/database/schema/core/defs.js";
+import { portfolioSnapshots } from "@/database/schema/trading/defs.js";
+
+export class ObjectIndexService {
+  /**
+   * Populate object_index with trade data
+   */
+  async populateTrades(competitionId?: string) {
+    console.log(`Populating trades${competitionId ? ` for competition ${competitionId}` : ''}`);
+    
+    const query = competitionId 
+      ? db.query.trades.findMany({ 
+          where: eq(trades.competitionId, competitionId),
+          orderBy: (trades, { desc }) => [desc(trades.timestamp)]
+        })
+      : db.query.trades.findMany({
+          orderBy: (trades, { desc }) => [desc(trades.timestamp)]
+        });
+
+    const tradesToSync = await query;
+    
+    if (tradesToSync.length === 0) {
+      console.log('No trades to sync');
+      return;
+    }
+
+    const entries = tradesToSync.map(trade => ({
+      id: crypto.randomUUID(),
```

Comment:
we should `import { v4 as uuidv4 } from "uuid";` and use `uuidv4()` instead of `crypto.randomUUID()`